src/base/lib/Convex/Utxos.hs

```diff
@@ -564,22 +574,72 @@ extract :: (C.TxIn -> C.InAnyCardanoEra (C.TxOut C.CtxTx) -> Maybe a) -> Maybe A
 extract ex cred state = DList.toList . \case
   -- FIXME (koslambrou) why is this only extracting from babbage?
   BlockInMode C.BabbageEra block -> extractBabbage ex state cred block
+  BlockInMode C.ConwayEra  block -> extractConway ex state cred block
   _                              -> mempty
 
 {-| Extract from a block the UTXO changes at the given address
 -}
 extract_ :: AddressCredential -> UtxoSet C.CtxTx () -> BlockInMode -> UtxoChange C.CtxTx ()
 extract_ a b = foldMap fromEvent . extract (\_ -> const $ Just ()) (Just a) b
 
-{-| Extract from a transaction the UTXO changes at the given address}
- -}
-extractBabbageTxn
+extractConway
   :: (C.TxIn -> C.InAnyCardanoEra (C.TxOut C.CtxTx) -> Maybe a)
+  -> UtxoSet C.CtxTx a
   -> Maybe AddressCredential
+  -> Block ConwayEra
+  -> DList (UtxoChangeEvent a)
+extractConway ex state cred (CS.Block _blockHeader txns) = foldMap (extractConwayTxn' ex state cred) txns
+
+extractConwayTxn'
+  :: forall a. (C.TxIn -> C.InAnyCardanoEra (C.TxOut C.CtxTx) -> Maybe a)
   -> UtxoSet C.CtxTx a
-  -> C.Tx BabbageEra
-  -> [UtxoChangeEvent a]
-extractBabbageTxn ex cred state = DList.toList . extractBabbageTxn' ex state cred
+  -> Maybe AddressCredential
+  -> C.Tx ConwayEra
+  -> DList (UtxoChangeEvent a)
+extractConwayTxn' ex UtxoSet{_utxos} cred theTx@(Tx txBody _) =
```

Comment:
Seems like a lot of duplication with `extractBabbageTxn'. Have you checked if you could write a single for that?

Reply:
You're right, I'll have a look